File: doubly_linked_list.c
#include<stdio.h>
#include<stddef.h>//importing library
struct node{//structure of the node in list
int data;
struct node *next;
struct node *prev;
};
void printforward(struct node *head);//printing the list in forward direction
void printbackward(struct node *item);//printing the list in the backward direction
struct node *head=NULL;//defining a head
struct node *iter=NULL;//defining a iter for the printing and assigning the value of the node
int main(){
    int counter=1;
    printf("enter the values\n");
    while(1==1){
        struct node *list=(struct node*)malloc(sizeof(struct node));//generating the node
        if(counter==1){
                //checking the head condition

                head=list;
                //creating the reference to the first node in the list
        head->next=NULL;//removing the garbage values from the pointers
            scanf("%d",&head->data);
            head->prev=NULL;//assigning the null to the previous pointer
        iter=head;
        counter++;//removing the head condition
        printf("\n");
        printforward(head);//printing the list
        }
        else{
                scanf("%d",&list->data);//value scanned in the node
                list->next=NULL;
            iter->next=list;//assigning the address of the node to the previous node
            list->prev=iter;//assigning  and storing the address of the previous node in the current node
            iter=list;//creating the new reference of the current node
        printf("\n");
            printforward(head);//printing the list in forward direction
            printbackward(list);//printing the list in the backward direction


        }

    }



}
void printforward(struct node *head){
while(head->next!=NULL){
    printf("%d ",head->data);
head=head->next;

}
printf("%d\n",head->data);
}

void printbackward(struct node *item){
do{
    printf("%d ",item->data);
    item=item->prev;

}while(item->prev!=NULL);
printf("%d\n",item->data);

}

File: doubly_linked_list.java
import java.util.Scanner;//importing scanner for user input

public class doubly_linked_list {
    public static void main(String[] args){
        Scanner scan=new Scanner(System.in);//creating the object for scanner
        System.out.println("enter the values");
        node head=new node();//creating the head node for the linked list
        node iter=null;//defining a reference variable
        print printer=new print();//defining an object to print the linked list

        int counter=1;//checking the condition to be the head
        while(true){
            node nextitem=new node();//generating node in the list
            if(counter==1){//going for head condition
                head.data=scan.nextInt();//scanning data to the head
                head.prev=null;//head is first element so its previous element in the list is null
                iter=head;//creating a reference variable for the head
                printer.printforward(head);//printing the list
                counter++;//countering the condition for the element to be the head
            }
            else{
                nextitem.data=scan.nextInt();//scanning the data in the node of the list
                iter.next=nextitem;//storing the address of the new node in the previous node
                nextitem.prev=iter;//storing the previous element address in the new node
                iter=nextitem;//creating the reference of the new node in the list
                printer.printforward(head);//printing the list by iterating the list in the forward direction
                System.out.println(" ");
                printer.printbackward(nextitem);//printing the list backward;



            }
        }
    }
}
class node{//class which defines the node
    int data;
    node next;
    node prev;
    node(){
        next=null;
        prev=null;
    }
}
class print{//class to print the list in forward as well as backward direction
    void printforward(node head){
        while(head.next!=null){
            System.out.printf("%d ",head.data);
            head=head.next;
        }
        System.out.printf("%d\n",head.data);
    }
    void printbackward(node tail){
        do{
            System.out.printf("%d ",tail.data);
            tail=tail.prev;

        }while(tail.prev!=null);
        System.out.printf("%d\n",tail.data);



    }
}
